packages/react-google-maps-api/src/components/dom/OverlayView.tsx

```diff
@@ -161,35 +150,32 @@ export class OverlayView extends React.PureComponent<OverlayViewProps, OverlayVi
     const boundsString = convertToLatLngBoundsString(this.props.bounds)
 
     if (prevPositionString !== positionString || prevBoundsString !== boundsString) {
-      setTimeout(() => {
-        this.state.overlayView !== null && this.state.overlayView.draw()
-      }, 0)
+      this.overlayView.draw()
+    }
+    if (prevProps.mapPaneName !== this.props.mapPaneName) {
+      this.updatePane()
     }
   }
 
   componentWillUnmount(): void {
-    if (this.state.overlayView !== null) {
-      if (this.props.onUnmount) {
-        this.props.onUnmount(this.state.overlayView)
-      }
-
-      this.state.overlayView.setMap(null)
-    }
+    this.overlayView.setMap(null)
   }
 
   render(): React.ReactPortal | React.ReactNode {
-    return this.mapPaneEl ? (
-      ReactDOM.createPortal(
-        <div ref={this.containerRef} style={{ ...this.state.containerStyle, position: 'absolute' }}>
-          <ContentMountHandler onLoad={this.setOverlayViewCallback}>
-            {React.Children.only(this.props.children)}
-          </ContentMountHandler>
+    const paneEl = this.state.paneEl
+    if (paneEl) {
+      return ReactDOM.createPortal(
+        <div
+          ref={this.containerRef}
+          style={{ ...this.state.containerStyle, position: 'absolute' }}
```

Comment:
please extract to useMemo

Reply:
may be better use className instead of style. I'm working on CSP, and most secure settings require to disable inline-styles.
Currently inline styles is legacy from other projects. Not requirement, just a notice.

Reply:
Is difficult to set dynamic style with className, if you know how to do that, please let me know